collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java

```diff
@@ -959,14 +930,20 @@ private String getFileExtensionFromUri(Uri fileUri) {
         }
     }
 
-    private QuestionWidget getWidgetWaitingForBinaryData() {
+    public QuestionWidget getWidgetWaitingForBinaryData() {
         QuestionWidget questionWidget = null;
-        for (QuestionWidget qw : ((ODKView) currentView).getWidgets()) {
-            if (qw.isWaitingForData()) {
-                questionWidget = qw;
+
+        if (currentView != null) {
```

Comment:
I think it would be better to check for null after casting, I mean:
```
ODKView odkView = (ODKView) currentView;
if (odkView != null) {
...
}
```

Reply:
Okay @grzesiek2010, I will make that change. Thanks for guiding me:)